share/wake/lib/core/list.wake

```diff
@@ -845,19 +844,99 @@ export def groupBy (cmp: a => a => Boolean) (list: List a): List (List a) =
         _ = (elem, Nil), acc
     foldr combo Nil sorted
 
-export def lookup (selectFn: a => Boolean): List (Pair a b) => List b =
+# Given a list of pairs, find all right elements such that
+# their left element matches a selection function. This
+# can be used to lookup values by keys if the list
+# represents a dictionary for instance.
+#
+# Parameters:
+#  - `selectFn`: The function that selects which keys to filter by
+#  - `list`: The list of key-value pairs to pull selected values out of
+#
+# Examples:
+# ```
+#   def l = ("Alice" :-> 85, "Bob" :-> 4, "Carol" :-> 20, "Bob" :-> 15, Nil)
+#   lookup (_==~"Alice") l = 85, Nil
+#   lookup (_==~"Bob") l = 4, 15, Nil
+#   lookup (_==~"Mal") l = Nil
+#   lookup (\_ True) l = 85, 4, 20, 15, Nil
+#   lookup (\_ False) l = Nil
+# ```
+export def lookup (selectFn: a => Boolean): (list: List (Pair a b)) => List b =
   def select (Pair key value) = if selectFn key then Some value else None
   mapPartial select
 
-export def editLookup (selectFn: a => Boolean) (editFn: b => b): List (Pair a b) => List (Pair a b) =
+# Given a list of pairs, edit all right elements such that
+# their left element matches a selection function. This
+# can be used to eidts values in a dictionary for instance.
+#
+# Parameters:
+#  - `selectFn`: The function that selects which keys to edit the values of
+#  - `editFn`: The function used to update values
+#  - `list`: The list of key-value pairs to pull selected values out of
+#
+# Examples:
+# ```
+#   def l = ("Alice" :-> 85, "Bob" :-> 4, "Carol" :-> 20, "Bob" :-> 15, Nil)
+#   editLookup (_==~"Alice") (_-5) l = "Alice" :-> 80, "Bob" :-> 4, "Carol" :-> 20, "Bob" :-> 15, Nil
+#   editLookup (_==~"Bob") (_-5) l = "Alice" :-> 85, "Bob" :-> -1, "Carol" :-> 20, "Bob" :-> 10, Nil
+#   editLookup (_==~"Mal") (_+1000) l = "Alice" :-> 85, "Bob" :-> 4, "Carol" :-> 20, "Bob" :-> 15, Nil
+#   editLookup (\_ True) (_+1) l = "Alice" :-> 86, "Bob" :-> 5, "Carol" :-> 21, "Bob" :-> 16, Nil
+#   editLookup (\_ False) (+1) l = "Alice" :-> 86, "Bob" :-> 5, "Carol" :-> 21, "Bob" :-> 16, Nil
```

Comment:
You should actually run these examples and just copy-paste the anwer. The above example is wrong.